### PR TITLE
Exclude `git fetch .` reflog events when inferring fork point + update deps

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -37,17 +37,22 @@ RUN set -x \
...
+  && apt-get update \
+  && apt-get install --no-install-recommends -y nodejs \
+  && rm -rf /var/lib/apt/lists/* \
+  && curl -L https://npmjs.org/install.sh | sh \
+  && npm install --global doctoc remark-cli remark-validate-links
+
 # Secondary packages needed in just one (or few) steps of the pipeline:
 # (package       => needed for command(s))
 # binutils       => strings
-# netcat         => nc
```
Comment (PawelLipski) [inline Dockerfile:43]: No longer used once the `scripts/run-ui-tests` has been simplified

File: .circleci/config.yml [hunk 2]
```diff
@@ -2,20 +2,24 @@ version: 2.1
...
+      - image: 'gitmachete/intellij-plugin-ci:4.0.0'
     working_directory: ~/git-machete-intellij-plugin
     steps:
       - checkout
       - run:
           name: Run pre-compile checks
           command: ./scripts/run-pre-build-checks
       - run:
-          name: Validate Markdown links
+          name: Validate Markdown ToCs
```
Comment (mkondratek) [inline .circleci/config.yml:13]: +1

**Review by mkondratek — APPROVED** [review]

**Review by Germanstreetstyle — CHANGES_REQUESTED** [review]
- [ ]

**Comment by PawelLipski** [comment]
❓

**Pull request merged**